compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -488,7 +488,7 @@ object SymDenotations {
           if kind == FlatName && !encl.is(JavaDefined) then qn.compactified else qn
         val fn = name replace {
           case name: SimpleName => qualify(name)
-          case name @ AnyQualifiedName(_, _) => qualify(name.mangled.toSimpleName)
```

Comment:
In fact I now doubt whether we should have this case at all. If we remove this line, then #15702 succeeds. So if we can to this, it would be better.